src/pages/QuestionPage.tsx

```diff
@@ -4,25 +4,68 @@ import Question from "../components/Question";
 import Answer from "../components/Answer";
 import { useEffect, useState } from "react";
 import { usePollForGetQuestion } from "../hooks/usePollForGetQuestion";
-import { useAppSelector } from "../state/reduxHooks";
+import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
+import { submitQuestion, setQuestionAnswer } from "../state/questionSlice";
+import { setGotQuestion, setRequestNextQuestion } from "../state/gameSlice";
+import { useSubmitAnswer } from "../hooks/useSubmitAnswer";
+import { useNextQuestion } from "../hooks/useNextQuestion"
 
 export default function QuestionPage() {
 
   const { prompt, answers, index } = useAppSelector(state => state.question);
-  const [answer, setAnswer] = useState<string | undefined>(undefined);
+  const { gotQuestion } = useAppSelector(state => state.game);
+  const { isHost } = useAppSelector(state => state.player);
+  const [answer, setAnswer] = useState<number | undefined>(undefined);
+  const [hasCurrentQuestion, setHasCurrentQuestion] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
 
   usePollForGetQuestion();
+  useSubmitAnswer();
+  useNextQuestion();
 
-  const submitAnswer = () => {
+  const submitAnswer = (event:React.MouseEvent) => {
+    if (answer) {
```

Comment:
This causes a bug (i.e. What if the index of the `answer` is 0? Then the answer won't be submitted since it will come to false). Should be:
```
if (answer !== undefined) {
```